### PR TITLE
Trainee can buy into a session

### PR DESCRIPTION
**Pull request by GergKllai1**
# PT Link

[Trainee can buy into a session](https://www.pivotaltracker.com/story/show/163530825)

# User Story

```
As a Trainee
In order to participate in training sessions
I would like to be able to pay using my credit card
```

# Screenshots

feb-04-2019 13-45-05

# Description of PR

- Adds stripe payment method
- Adds transaction model and controller with create and new action
- Refactors Sessions component to complement payment flow
- Adds sessions controller and show page with show action
- Passes down session information in stripe
- Adds error and success messages on payment

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,5 +1,7 @@
 Rails.application.routes.draw do
   devise_for :users, controllers: { omniauth_callbacks: :omniauth_callbacks }
   root controller: :main, action: :index
-  get '*path', to: 'main#index', via: :all
+	# get '*path', to: 'main#index', via: :all
```
Comment (magnus-thor) [inline config/routes.rb]: remove commented out code

File: app/assets/javascripts/application.js [hunk 2]
```diff
@@ -1,16 +1,42 @@
...
-// compiled file. JavaScript code in this file should be added after the last require_* statement.
-//
-// Read Sprockets README (https://github.com/rails/sprockets#sprockets-directives) for details
-// about supported directives.
-//
 //= require rails-ujs
 //= require activestorage
 //= require turbolinks
 //= require_tree .
+
```
Comment (HannaSwDn) [inline app/assets/javascripts/application.js:5]: remove empty line :)

File: client/bundles/Main/components/Main.jsx [hunk 3]
```diff
@@ -40,6 +40,8 @@ export default class Main extends React.Component {
 		})
 	}
 
+
```
Comment (HannaSwDn) [inline client/bundles/Main/components/Main.jsx]: remove two empty lines

File: config/application.rb [hunk 4]
```diff
@@ -23,6 +23,8 @@ class Application < Rails::Application
       generate.routing_specs false
       generate.controller_specs false
       generate.system_tests false
-    end
+		end
+		config.stripe.publishable_key = 'pk_test_QicERB8w3kyqaYW3hUUQylRH'
```
Comment (HannaSwDn) [inline config/application.rb:27]: Is it okay to add this to Github?

Comment (HannaSwDn) [reply config/application.rb:27]: just wanna be sure haha

**Review by johannaanden — APPROVED** [review]
lgtm

**Review by schneiderman805 — APPROVED** [review]
could not see anything else that needs to be changed, looks good

File: config/routes.rb (hunk 1 above)
Comment (GergKllai1) [reply config/routes.rb]: Thanks for highlighting it :) We modified it to represent out application flow

File: app/assets/javascripts/application.js (hunk 2 above)
Comment (GergKllai1) [reply app/assets/javascripts/application.js:5]: Done

File: client/bundles/Main/components/Main.jsx (hunk 3 above)
Comment (GergKllai1) [reply client/bundles/Main/components/Main.jsx]: Done

File: config/application.rb (hunk 4 above)
Comment (GergKllai1) [reply config/application.rb:27]: Yes, it is public key

**Review by HannaSwDn — APPROVED** [review]

**Pull request merged**